LifeSpace/Debug/LogViewer.swift

```diff
@@ -0,0 +1,112 @@
+//
+//  LogShareView.swift
+//  LifeSpace
+//
+//  Created by Vishnu Ravi on 11/8/24.
+//
+
+import OSLog
+import Spezi
+import SwiftUI
+
+
+struct LogViewer: View {
+    @Environment(LogManager.self) var manager
+    
+    @State private var startDate: Date = Calendar.current.date(byAdding: .day, value: -1, to: Date()) ?? Date()
+    @State private var endDate = Date()
+    @State private var selectedLogType: LogType = .all
+    @State private var logs = ""
+    @State private var isLoading = false
+    
+    var body: some View {
+        VStack {
+            /// Date range selection
+            HStack {
+                DatePicker("FROM", selection: $startDate, displayedComponents: .date)
+                Spacer()
+                DatePicker("TO", selection: $endDate, displayedComponents: .date)
+            }
+            .padding()
+            
+            /// Log type selection
+            Picker("LOG_TYPE", selection: $selectedLogType) {
+                ForEach(LogType.allCases) { type in
+                    Text(type.rawValue).tag(type)
+                }
+            }
+            .pickerStyle(SegmentedPickerStyle())
+            .padding()
+            
+            ScrollView {
+                if isLoading {
+                    ProgressView("LOADING_LOGS")
+                        .padding()
+                } else {
+                    Text(logs)
+                        .padding()
+                }
+            }
+        }
+        .navigationTitle("LOG_VIEWER")
+        .onAppear {
+            Task {
+                await queryLogs()
+            }
+        }
+        .onChange(of: startDate) {
+            Task {
+                await queryLogs()
+            }
+        }
+        .onChange(of: endDate) {
+            Task {
+                await queryLogs()
+            }
+        }
+        .onChange(of: selectedLogType) {
+            Task {
+                await queryLogs()
+            }
+        }
+        .toolbar {
+            if !logs.isEmpty, let fileURL = saveLogsToTextFile(logs) {
```

Comment:
`DataRepresentation` as an async export function you can use here to avoid saving a file to disk every time the view is re-rendered: https://developer.apple.com/documentation/CoreTransferable/DataRepresentation. It would also avoid creating a file which we currently don't clean up after the view is no longer shown. It should happen in a did disappear or after the share sheet has been done. A data representation should solve that.

Reply:
Tried to do it this way, but this results in the entire text being shared as a text blob. I would prefer to generate a file that the user can share (e.g. attach to an email), especially if the log is very long. I tried using the `suggestedFileName` attribute, but it still shares as a text blob.

Reply:
Will leave it as a text blob export for now.